crypto/keys/keybase_base.go

```diff
@@ -34,6 +46,30 @@ type (
 	}
 )
 
+// WithKeygenFunc applies an overridden key generation function to generate the private key.
+func WithKeygenFunc(f PrivKeyGenFunc) KeybaseOption {
+	return func(o *kbOptions) {
+		o.keygenFunc = f
+	}
+}
+
+// newBaseKeybase generates the base keybase with defaulting to tendermint SECP256K1 key type
+func newBaseKeybase(optionsFns ...KeybaseOption) baseKeybase {
+	// Default options for keybase
+	options := kbOptions{keygenFunc: baseSecpPrivKeyGen}
```

Comment:
why not an inline declaration such as the following?

```suggestion
	options := kbOptions{keygenFunc: func(bz [32]byte) tmcrypto.PrivKey { return secp256k1.PrivKeySecp256k1(bz) }}
```

Reply:
readability and to document what it does, would you like me to make this change?

Reply:
Not necessarily, it's just a personal preference of mine. You made a good point, I trust your judgement.